src/extension.ts

```diff
@@ -126,19 +137,28 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         }),
         registerCommand(CMD_RESTART_SERVER, async () => {
             await runServer();
+            await sendHeapEventWithMetadata(CMD_RESTART_SERVER);
         }),
         registerCommand(CMD_SELECT_ENV, async () => {
             const result = await selectEnvironment();
             runServer(result);
             if (result) {
                 statusBarItem.text = `$(kedro-logo)` + ' ' + result.label;
             }
+            await sendHeapEventWithMetadata(CMD_SELECT_ENV);
         }),
         registerCommand('pygls.server.executeCommand', async () => {
             await executeServerCommand(lsClient);
         }),
-        registerCommand('kedro.sendDefinitionRequest', async (word) => {
+        registerCommand(CMD_DEFINITION_REQUEST, async (word) => {
             await executeServerDefinitionCommand(lsClient, word);
+            await sendHeapEventWithMetadata(CMD_DEFINITION_REQUEST);
```

Comment:
Maybe we don't need to await this? @jitu5

Reply:
Yes.